src/diagram/components/quantity-node.scss

```diff
@@ -1,167 +1,358 @@
+          background-color: theme-var($--theme-color-5);
+          border: solid 1px $default-fg-color;
+          border-radius: 3px;
+          height: 22px;
+          padding: 2px 4px;
+          width: 100%;
+        }
+        &.unit { width: $input-half-width; }
+        &.value {
+          text-align: right;
+          width: $input-half-width;
+        }
+        &.static {
+          background-color: theme-var($--theme-color-7);
+          padding: 3px 6px;
+          &:hover { outline: none; }
+        }
+        &.no-value { color: gray; }
+        &.invalid { outline: 2px solid $invalid-red; }
       }
     }
+
+    .error-icon {
+      bottom: 2.5px;
+      height: 20px;
+      left: 3px;
+      position: absolute;
+      width: 20px;
+    }
     .error-message {
+      color: $invalid-red;
       display: inline-block;
+      font-size: 12px;
+      height: auto;
+      left: 0;
       overflow-wrap: normal;
-      width: calc($input-half-width + 8px);
-      vertical-align: top;
-      margin-top: 3px;
-      margin-left: 3px;
+      position: absolute;
+      top: 100%;
+      width: 100%;
+
+      p { margin: 0; }
     }
   }
 
   .variable-info-floater {
+    bottom: 4px;
+    cursor: pointer;
+    height: 20px;
     position: absolute;
-    right: 5px;
-    bottom: 5px;
-    width: 18px;
-    height: 18px;
-  }
-  .variable-info-floater svg {
-    width: 18px;
-    height: 18px;
-    fill: white;
-  }
+    right: 4px;
+    width: 20px;
 
-  select, input {
-    background-color: white;
-    padding: 2px 6px;
-    border: 0 solid #999;
+    button {
+      background: theme-var($--theme-color-5);
+      border: none;
+      border-radius: 50%;
+      cursor: pointer;
+      display: block;
+      height: 20px;
+      margin: 0;
+      padding: 0;
+      width: 20px;
+
+      &:hover { background-color: theme-var($--theme-color-4); }
+      &:active {
+        background-color: theme-var($--theme-color-2);
+        svg {
+          .background {
+            fill: theme-var($--theme-color-5);
+            stroke: theme-var($--theme-color-5);
+          }
+          .foreground {
+            fill: white;
+            stroke: white;
+          }
+        }
+      }
+
+      svg {
+        .background {
+          fill: theme-var($--theme-color-5);
+          stroke: theme-var($--theme-color-5);
+        }
+        .foreground {
+          fill: theme-var($--theme-color-1);
+          stroke: theme-var($--theme-color-1);
+        }
+      }
+    }
   }
 
-  .variable-description-area {
-    resize: none;
-    width: 100%;
-    text-align-last:left!important;
-    border-radius: 5px;
-    border: none;
-    margin: 3px;
-    padding-left: .5em;
-    padding-right: .5em;
-    overflow-y: scroll;
+  .flow-handle {
+    &:hover {
+      background-color: $hover-blue !important;
+      box-sizing: content-box !important;
+      cursor: pointer;
+    }
+    &:active {
+      background-color: $select-blue !important;
```

Comment:
I agree using `!important` is not a good practice generally. Some of the modules we're using use inline styles that aren't easy to modify, however. I've found that using `!important` to override those styles is sometimes necessary.

In the specific case of these node handles, I found that dynamically changing the style values set for them in quantity-node.tsx would break their functionality. For example, to change the background color when you click on a handle, you have to add something like `onMouseDown={changeBGColor}` but that `changeBGColor` function interrupts and breaks the drag functionality that you need to connect the node to another node.

There may be some other way around that that I'm unaware of, but I suspect it wouldn't be easy.

For now I'll add some comments about the use of `!important`.